File: item-list/src/main/java/com/gurunars/item_list/SelectableItemViewBinderString.java
package com.gurunars.item_list;

import android.graphics.Color;
import android.support.annotation.Nullable;
import android.widget.TextView;

class SelectableItemViewBinderString<ItemType extends Item> extends ItemViewBinderString<SelectableItem<ItemType>> {

    @Override
    public void bind(TextView itemView, SelectableItem<ItemType> item, @Nullable SelectableItem<ItemType> previousItem) {
        super.bind(itemView, item, previousItem);
        itemView.setBackgroundColor(item.isSelected() ? Color.RED : Color.TRANSPARENT);
    }

}

File: item-list/src/main/java/com/gurunars/item_list/SelectableItemViewBinderString.kt
package com.gurunars.item_list

import android.graphics.Color
import android.widget.TextView

internal class SelectableItemViewBinderString<ItemType : Item> : ItemViewBinderString<SelectableItem<ItemType>>() {

    override fun bind(itemView: TextView, item: SelectableItem<ItemType>, previousItem: SelectableItem<ItemType>?) {
        super.bind(itemView, item, previousItem)
        itemView.setBackgroundColor(if (item.isSelected) Color.RED else Color.TRANSPARENT)
    }

}
